Build error with gcc 12

## System information
Arch Linux x86_64 (cpu/gpu is not relevant here, since it's a build error).

## Issue behavior
### Describe the current behavior

I'm getting the following build error when using gcc 12.1.0:

```
In file included from /build/intel-media-sdk-git/src/MediaSDK/tests/unit/suites/mfx_dispatch/linux/mfx_dispatch_test_main.cpp:21:
/build/intel-media-sdk-git/src/MediaSDK/tests/unit/suites/mfx_dispatch/linux/mfx_dispatch_test_main.h:47:13: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
   47 | extern std::unique_ptr<MockCallObj> g_call_obj_ptr;
      |             ^~~~~~~~~~
/build/intel-media-sdk-git/src/MediaSDK/tests/unit/suites/mfx_dispatch/linux/mfx_dispatch_test_main.h:30:1: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
   29 | #include <list>
  +++ |+#include <memory>
   30 | #include <algorithm>
/build/intel-media-sdk-git/src/MediaSDK/tests/unit/suites/mfx_dispatch/linux/mfx_dispatch_test_main.h: In function ‘void ResetMockCallObj()’:
/build/intel-media-sdk-git/src/MediaSDK/tests/unit/suites/mfx_dispatch/linux/mfx_dispatch_test_main.h:64:5: error: ‘g_call_obj_ptr’ was not declared in this scope
   64 |     g_call_obj_ptr.reset(new MockCallObj);
      |     ^~~~~~~~~~~~~~
```

The same error also occurs in other source files. Complete [build log](https://paste.ee/p/nZdAL) (expires in one year).

It builds fine with gcc 11.3.0.

Build options:

```
cmake -B build -S MediaSDK-intel-mediasdk-22.4.2 \
        -DBUILD_ALL:BOOL='ON' \
        -DBUILD_TOOLS:BOOL='ON' \
        -DCMAKE_BUILD_TYPE:STRING='None' \
        -DCMAKE_INSTALL_PREFIX:PATH='/usr' \
        -DENABLE_ITT:BOOL='OFF' \
        -DENABLE_OPENCL:BOOL='OFF' \
        -DENABLE_WAYLAND:BOOL='ON' \
        -DENABLE_X11_DRI3:BOOL='ON' \
        -DMFX_APPS_DIR='/usr/lib/mfx' \
        -Wno-dev
```

### Describe the expected behavior
The build process to complete successfully without errors.

## Debug information
- What's libva/libva-utils/gmmlib/media-driver/Media SDK version?
  - libva 2.14.0
  - libva-utils 2.14.0
  - gmmlib 22.1.3
  - media-driver 22.4.2
  - media-sdk 22.4.2
- Do you want to contribute a PR? (yes/no):
  - yes

## Solution

As the gcc informative message suggests, this can be fixed by including the `<memory>` C++ header.

See documentation about [porting to gcc 12](https://gcc.gnu.org/gcc-12/porting_to.html).

--- a/tests/unit/suites/mfx_dispatch/linux/mfx_dispatch_test_main.h
+++ b/tests/unit/suites/mfx_dispatch/linux/mfx_dispatch_test_main.h
@@ -26,6 +26,7 @@
 
 #include <gtest/gtest.h>
 #include <map>
+#include <memory>
 #include <list>
 #include <algorithm>
 

--- a/tests/unit/suites/tracer/linux/mfx_tracer_test_main.h
+++ b/tests/unit/suites/tracer/linux/mfx_tracer_test_main.h
@@ -26,6 +26,7 @@
 
 #include <gtest/gtest.h>
 #include <map>
+#include <memory>
 #include <list>
 #include <algorithm>
 

